src/common/imageio_im.c

```diff
@@ -126,8 +126,10 @@ dt_imageio_retval_t dt_imageio_open_im(dt_image_t *img, const char *filename, dt
     MagickRelinquishMemory(profile_data);
   }
 
+  // As a warning to those who will modify the loader in the future:
+  // MagickWandTerminus() cannot be called on successful image reading.
+  // See https://github.com/darktable-org/darktable/issues/13090 regarding the consequences.
   DestroyMagickWand(image);
-  MagickWandTerminus();
```

Comment:
If so, should also then remove the matching `MagickWandGenesis()` at the start to avoid it creating a new environment on every file open?

Isn't a full fix to call matching `MagickWandGenesis()`/`MagickWandTerminus()` at the start/end of dt session then?

Reply:
> If so, should also then remove the matching MagickWandGenesis() at the start to avoid it creating a new environment on every file open?

You added calls to MagicWandGenesis()/MagickWandTerminus() functions in 6f5e4980. Before this, the loader was working fine without these calls, although it's hard for me to say if such usage is correct/optimal. In any case, ImageMagick function calls worked even without creating an environment (whatever that term means).

In summary, the code works both without a MagicWandGenesis call and with a MagicWandGenesis, but without the corresponding explicit MagickWandTerminus.

I don't know which of these options is "more correct".

> Isn't a full fix to call matching MagickWandGenesis()/MagickWandTerminus() at the start/end of dt session then?

Even when we won't use ImageMagick in this session?

Reply:
Yes, that's why I sad we should remove them together as a matching pair. I just followed the IM API usage example, I'm not sure what they do exactly either, sorry 😯